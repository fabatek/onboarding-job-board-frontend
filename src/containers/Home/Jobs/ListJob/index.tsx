import Job from '../../../model/job';
export default function ListJob(props:{currentJobs:Job[]}) {
    const {currentJobs} = props
    return (
        <div>
            {currentJobs?.map((item: Job, key: number) => {
                console.log(item.posted)
                return (
                    <li key={key} className={`list-group-item d-flex job`} data-testid="test100Jobs">
                        <img className='job__img' alt='logo' src={item.img} />
                        <div className='job__tittle'>
                            <h4>{item.jobName}</h4>
                            <ul className='job__tech'>
                                <li>Html</li>
                                <li>Css</li>
                                <a href="/#">Sign in to view</a>
                            </ul>
                        </div>
                        <div className='job__city'>
                            <p>{item.city}</p>
                            <p className='job__posted'>{item.posted.getHours}</p>
                        </div>
                    </li>
                )
            })}
        </div>
    )
}

