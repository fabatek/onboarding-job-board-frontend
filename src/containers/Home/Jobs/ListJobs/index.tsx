import Job from '../../../model/job';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faDollarSign, faCaretRight } from '@fortawesome/free-solid-svg-icons';

export default function ListJobs(props: { currentJobs: Job[] }) {
    const { currentJobs } = props

    return (
        <ul className='container-fluid mb-2' id='listJobs'>
            {currentJobs.map((job: Job, key: number) => {
                const hours = Math.floor(Math.random() * 60) + 1
                const city = ['Ho Chi Minh', 'Ha Noi', 'Da Nang']
                const indexCity = Math.floor(Math.random() * 3)
                return (
                    <li key={key} className={`list-group-item d-flex job`} data-testid="test100Jobs">
                        <img className='job__img' alt='logo' src={job.img} />
                        <div className='job__tittle'>
                            <h4>{job.jobName}</h4>
                            <div className='job__tech'>
                                <a href="/#"><FontAwesomeIcon className='job__icon' icon={faDollarSign} /> Sign in to view <FontAwesomeIcon icon={faCaretRight} /></a>
                                <div>
                                    <button>Css</button>
                                    <button>Html</button>
                                    <button>Javascript</button>
                                    <button>Fresher accept</button>
                                </div>
                            </div>
                        </div>
                        <div className='job__city'>
                            <p>{job.city}</p>
                            <p className='job__posted'>{hours}m</p>
                        </div>
                    </li>
                )
            })}
        </ul>
    )
}

