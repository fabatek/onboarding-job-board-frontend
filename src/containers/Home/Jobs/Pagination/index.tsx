import { useEffect, useState } from "react"
import { useSelector } from "react-redux";
import Job from "../../../model/job";
import { RootState } from "../../../Redux/store/store";

export default function Pagination(props: { jobsPerPage: number, totalJobs: number, paginate: any, currentPage: number,totalSearchJob: number }) {

    const { jobsPerPage, totalJobs, paginate, currentPage, totalSearchJob } = props
    const [pageNumber, setPageNumber] = useState<Number[]>([]);
    const search = useSelector((state:RootState)=> state.search)
    let totalPage = 0;
    search.search == "" ? totalPage = Math.ceil(totalJobs / jobsPerPage) : totalPage = Math.ceil(totalSearchJob / jobsPerPage)

    useEffect(() => {
        const listPageNumber = []
        for (let i = 1; i <= totalPage; i++) {
            listPageNumber.push(i)
        }
        setPageNumber(listPageNumber);
    }, [totalPage])

    return (
        <>
            {pageNumber.map((number, index) => {
                return (
                    <li className='page-item pagination__item' data-testid="pagination" key={index} >
                        <a onClick={() => {
                            paginate(number)
                        }} href="/#" className={`page-link pagination__item__link ${number === currentPage ? "pagination__item--active" : ""}`}>
                            {number}
                        </a>
                    </li>
                )
            })}
        </>
    )
}
