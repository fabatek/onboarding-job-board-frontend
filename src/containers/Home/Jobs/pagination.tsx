import React from "react";
const Pagination =(jobPerPage:number, totalJobs: number,paginate:any ) => {
    const pageNumber = [];

    for (let i = 1; i <= Math.ceil(totalJobs / jobPerPage); i++) {
        pageNumber.push(i);
    }
    return pageNumber?.map((number) => {
        return (
            <li className='page-item' key={number}>
                <a onClick={()=>paginate(number)} href='#listJobs' className='page-link'>
                    {number}
                </a>
            </li>
        )
    })
}
export default Pagination