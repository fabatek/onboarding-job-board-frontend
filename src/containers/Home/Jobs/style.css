.job-tittle{
    margin-left: 50px;
}
.job-city{
    position: absolute;
    top: 40px;
    right: 20px;
    content: "";
}
.job-content-active{
    border: solid 1px red;
}