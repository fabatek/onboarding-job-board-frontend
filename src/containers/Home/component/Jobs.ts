export interface Job {
    id: number;
    available: boolean;
    jobName: string;
    company: string;
    tags: string;
    jobType: string;
    location: string;
    salary: number;
    title: string;
}