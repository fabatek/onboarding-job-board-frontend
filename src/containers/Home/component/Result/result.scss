.result_container {
    h4 {
        margin-top: 10px;
        margin-bottom: 10px;

        line-height: 35px;
        font-size: 27px;
        font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
    }
    .result_detail {
        margin-top: 20px;
        .card_container {
            margin-top: 10px;
            .card {
                height: 450px;
                overflow: hidden;
                background-color: rgba($color: #c4c1c1, $alpha: 0.2);
                p {
                    margin-bottom: 0;
                    font-size: 16px;
                }
                .card_btn-detail, .card_btn-expired {
                    position:absolute;
                    bottom: 10px;
                }
                .card_btn-detail:hover, .card_btn-expired:hover {
                    width: 70%;
                }
            }
            .card:hover {
                box-shadow: 
                rgba(50, 50, 93, 0.25) 0px 13px 27px -5px, 
                rgba(0, 0, 0, 0.3) 0px 8px 16px -8px;;
            }
        }
    }
}