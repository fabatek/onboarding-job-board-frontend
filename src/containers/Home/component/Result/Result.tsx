import React from "react";
import Logo from "../../../../logo.svg";
import "./result.scss";
import { useAppSelector } from '../../../redux/hook'
import { Job } from '../Jobs'

interface props {
    job:Job
}

export const Box: React.FC<props> = ({
    job
}) => {
    return (
        <div className="col-sm-6 col-md-4 col-lg-3 col-xl-3 card__container" data-testid='job_result'>
            <div className="card">
                <img className="card-img-top" src={Logo} alt="Card img" />
                <div className="card-body">
                    <h5 className="card-title">{job.title}</h5>
                    <p className="card-text">
                        {job.company} is looking for {job.jobType}
                    </p>
                    <p className="card-text">
                        <b>Location:</b> {job.location}
                    </p>
                    <p className="card-text">
                        <b>Salary</b>: Up to {`${Math.round(job.salary * 10)}`}$
                    </p>
                    {job.available ? 
                    (
                        <button className="btn btn-primary card__btn--detail">Read more</button>
                    ) 
                    : 
                    (
                        <button className="btn btn-danger card__btn--expired">Expired</button>
                    )}
                </div>
            </div>
        </div>
  );
};

const Result = () => {
    const totalJobs = useAppSelector((state) => state.jobs.totalJobList)

    return (
        <div className="container result__container">
            <h4>Detail Jobs</h4>
            <div className="row result__detail">
                {totalJobs ? 
                (
                    totalJobs.slice(0, 100).map((job, index) => {
                        return (
                        <Box
                            key={index}
                            job={job}
                        />
                        );
                    })
                )
                :
                (
                    <div>
                        <h3>No Jobs Found</h3>
                    </div>
                )
                }
            </div>
        </div>
    );
};

export default Result;
