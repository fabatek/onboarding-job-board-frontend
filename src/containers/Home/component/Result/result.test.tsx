import React from "react";
import Result, { Box } from "./Result";
import { Job } from '../Jobs';
import { render, screen } from "@testing-library/react";
import { shallow } from 'enzyme'

it('render correctly when no jobs', () => {
    const fakeJobs:Job[] = []
    render(<Box job={fakeJobs[0]}/>)
    const tree = screen.getByTestId('job_result')
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveTextContent('No Jobs Found')
})

it('render correctly when there are some jobs', () => {
    const fakeJobs:Job[] = [
        {
            id: 1,
            available: true,
            jobName: 'dev',
            company: 'FabaTechnology',
            tags: 'IT, Front-end',
            jobType: 'full-time',
            location: 'Ho chi Minh City',
            salary: 450,
            title: 'Looking for fresher ReactJs Developer'
        },
        {
            id: 2,
            available: false,
            jobName: 'dev',
            company: 'FabaTechnology',
            tags: 'IT, Front-end',
            jobType: 'full-time',
            location: 'Ho chi Minh City',
            salary: 450,
            title: 'Looking for fresher ReactJs Developer'
        }
    ]
    render(
    <Box job={fakeJobs[1]}/>)
    const tree = screen.getByTestId('job_result')
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveTextContent('FabaTechnology')
    expect(tree).toHaveTextContent('dev')
    expect(tree).toHaveTextContent('Ho Chi Minh city')
    expect(tree).toHaveTextContent('Looing for fresher ReactJs Developert')
})

test('should render result with content', () => {
    render(<Result/>);
    const resultElement = screen.getByTestId('job_result')
    expect(resultElement).toBeInTheDocument()
    expect(resultElement).toHaveTextContent('is looking for')
    expect(resultElement).toHaveTextContent('Location:')
    expect(resultElement).toHaveBeenCalledTimes(1)
})

describe('pagination', () => {
    it('start with page 1', () => {
        const wrapper = shallow(<Result/>);
        const text = wrapper.find('button.result__pagination__btn:nth-child(2)').text()
        expect(text).toEqual(1)
    });
    it('Is page 1 active', () => {
        const wrapper = shallow(<Result/>);
        const text = wrapper.find('button.result__pagination__btn:nth-child(2)')
        expect(text.is('result__pagination__btn--active')).toBe(true)
    })
    it('increased and decreased current page by 1 when clicking next', () => {
        const wrapper = shallow(<Result/>);
        const increasedPageBtn = wrapper.find('button.result__pagination__btn:last-child')
        const decreasedPageBtn = wrapper.find('button.result__pagination__btn:first-child')
        increasedPageBtn.simulate('click')
        const page = wrapper.state('currentPage')
        expect(page).toEqual(2)
        decreasedPageBtn.simulate('click')
        expect(page).toEqual(1)
    });
    it('is prev btn disable when in page 1', () => {
        const wrapper = shallow(<Result/>)
        const text = wrapper.find('button.result__pagination__btn:first-child')
        expect(text.is('result__pagination__btn--disable')).toBe(true)
    })
})