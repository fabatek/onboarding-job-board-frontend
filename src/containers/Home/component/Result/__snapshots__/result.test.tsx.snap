// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly when no jobs 1`] = `
<div
  class="col-sm-6 col-md-4 col-lg-3 col-xl-3 card_custom"
  data-testid="job_result"
>
  <div
    class="card"
  >
    <img
      alt="Card img"
      class="card-img-top"
      src="logo.svg"
    />
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      />
      <p
        class="card-text"
      >
        
         is looking for 
        
      </p>
      <p
        class="card-text"
      >
        <b>
          Location:
        </b>
         
        
      </p>
      <p
        class="card-text"
      >
        <b>
          Salary
        </b>
        : Up to 
        0
        $
      </p>
      <button
        class="btn btn-danger btn_position"
      >
        Expired
      </button>
    </div>
  </div>
</div>
`;

exports[`render correctly when there are some jobs 1`] = `
<div
  class="col-sm-6 col-md-4 col-lg-3 col-xl-3 card_custom"
  data-testid="job_result"
>
  <div
    class="card"
  >
    <img
      alt="Card img"
      class="card-img-top"
      src="logo.svg"
    />
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Looking for fresher ReactJs Developer
      </h5>
      <p
        class="card-text"
      >
        FabaTechnology
         is looking for 
        dev
      </p>
      <p
        class="card-text"
      >
        <b>
          Location:
        </b>
         
        Ho chi Minh City
      </p>
      <p
        class="card-text"
      >
        <b>
          Salary
        </b>
        : Up to 
        4500
        $
      </p>
      <button
        class="btn btn-danger btn_position"
      >
        Expired
      </button>
    </div>
  </div>
</div>
`;
